### PR TITLE
Add a check for undesirable MNK skills

### PR DESCRIPTION
**Pull request by ayyaruq**
This could probably be a core module tbh, seems relevant to NIN and BLM too.

Rolling in a minor bug fix to the suggestion on Earth's Reply to make the actual suggestion clearer. A more general rework of Earth's Reply needs to happen in the future since there are times it's worth having for defence that shouldn't penalise users. It's currently minor so won't show for most players which is why I'm leaving it for now.

### CODE REVIEW

File: src/parser/jobs/mnk/modules/Undesirables.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+			icon: ACTIONS.ONE_ILM_PUNCH.icon,
+			content: <Fragment>
+				Avoid using <ActionLink {...ACTIONS.ONE_ILM_PUNCH} /> as you're losing uses of the higher potency <ActionLink {...ACTIONS.TRUE_STRIKE} /> and <ActionLink {...ACTIONS.TWIN_SNAKES} />.
+			</Fragment>,
+			tiers: {
+				1: SEVERITY.MAJOR,
+			},
+			value: oneIlmPunchCount,
+			why: <Fragment>
+				<ActionLink {...ACTIONS.ONE_ILM_PUNCH} /> cost {oneIlmPunchCount} more potent GCDs.
```
[the remark this answers is not in the document]
Comment (ayyaruq) [reply src/parser/jobs/mnk/modules/Undesirables.js]: In this case I'm going to leave it, but the Earth Tackle one is fixed.

File: src/parser/jobs/mnk/modules/GreasedLightning.js [hunk 2]
```diff
@@ -299,7 +294,20 @@ export default class GreasedLightning extends Module {
...
-		if (this._wastedWind > 0) {
+		if (missedEarth) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.RIDDLE_OF_EARTH.icon,
+				content: <Fragment>
+					Check the fight timeline to see when you can save <StatusLink {...STATUSES.GREASED_LIGHTNING_I} /> with <ActionLink {...ACTIONS.RIDDLE_OF_EARTH} />.
+				</Fragment>,
+				severity: SEVERITY.MINOR,
+				why: <Fragment>
+					<ActionLink {...ACTIONS.RIDDLE_OF_EARTH} /> was used {missedEarth} times without preserving <StatusLink {...STATUSES.GREASED_LIGHTNING_I} />,
```
[the remark this answers is not in the document]
Comment (ayyaruq) [reply src/parser/jobs/mnk/modules/GreasedLightning.js]: I pushed to the wrong branch, see #289

File: src/parser/jobs/mnk/modules/GreasedLightning.js [hunk 3]
```diff
@@ -299,7 +294,20 @@ export default class GreasedLightning extends Module {
 			}))
 		}
 
-		if (this._wastedWind > 0) {
+		if (missedEarth) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.RIDDLE_OF_EARTH.icon,
+				content: <Fragment>
+					Check the fight timeline to see when you can save <StatusLink {...STATUSES.GREASED_LIGHTNING_I} /> with <ActionLink {...ACTIONS.RIDDLE_OF_EARTH} />.
```
Comment (ackwell) [inline src/parser/jobs/mnk/modules/GreasedLightning.js]: This isn't really... a _suggestion_. It's not saying what they could do better, nor what they shouldn't be doing - it's just saying "LOOK AT SOMETHING ELSE".

I realise there's an existing one above, but I'd frankly much rather all instances of it be updated.

File: src/parser/jobs/mnk/modules/GreasedLightning.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ackwell) [reply src/parser/jobs/mnk/modules/GreasedLightning.js]: If you pushed it to that branch, it'll be missing the fix for this particular suggestion, 'cus it's new.

File: src/parser/jobs/mnk/modules/Undesirables.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	_onComplete() {
+		const oneIlmPunchCount = this._undesirables.filter(event => event.id === ACTIONS.ONE_ILM_PUNCH.id).length
+		const earthTackleCount = this._undesirables.filter(event => event.id === ACTIONS.EARTH_TACKLE.id).length
+		const lostTacklePotency = earthTackleCount * (ACTIONS.FIRE_TACKLE.potency - ACTIONS.EARTH_TACKLE.potency)
+
+		this.suggestions.add(new TieredSuggestion({
+			icon: ACTIONS.EARTH_TACKLE.icon,
+			content: <Fragment>
```
Comment (ackwell) [inline src/parser/jobs/mnk/modules/Undesirables.js]: Non-blocking, but `<Fragment>...</Fragment>` can (should?) be expressed as `<>...</>`. There's existing uses of `<Fragment>` in the codebase from before when `<>` was supported, but I'd like to ratchet the metric if possible.

**Pull request merged**